webroot/src/components/Licensee/LicenseeSearch/LicenseeSearch.ts

```diff
@@ -0,0 +1,180 @@
+        }
+
+        compactMemberStates.unshift(defaultSelectOption);
+
+        return compactMemberStates;
+    }
+
+    //
+    // Methods
+    //
+    initFormInputs(): void {
+        this.formData = reactive({
+            firstName: new FormInput({
+                id: 'first-name',
+                name: 'first-name',
+                label: computed(() => this.$t('common.firstName')),
+                placeholder: computed(() => this.$t('licensing.searchPlaceholderName')),
+                validation: Joi.string().min(0).max(10).messages(this.joiMessages.string),
+                enforceMax: true,
+            }),
+            lastName: new FormInput({
+                id: 'last-name',
+                name: 'last-name',
+                label: computed(() => this.$t('common.lastName')),
+                placeholder: computed(() => this.$t('licensing.searchPlaceholderName')),
+                validation: Joi.string().min(0).max(10).messages(this.joiMessages.string),
+                enforceMax: true,
+            }),
+            ssn: new FormInput({
+                id: 'ssn',
+                name: 'ssn',
+                label: computed(() => this.$t('licensing.ssn')),
+                placeholder: '000-00-0000',
+                validation: Joi.string().pattern(/^([0-9]){3}-([0-9]{2}-([0-9]{4}$))/).allow('').messages({
+                    ...this.joiMessages.string,
+                    'string.pattern.base': this.$t('inputErrors.ssnFormat'),
+                }),
+            }),
+            state: new FormInput({
+                id: 'state',
+                name: 'state',
+                label: computed(() => this.$t('common.state')),
+                valueOptions: this.stateOptions,
+            }),
+            submit: new FormInput({
+                isSubmitInput: true,
+                id: 'submit',
+            }),
+        });
+        this.watchFormInputs(); // Important if you want automated form validation
+    }
+
+    formatSsn(): void {
+        const { ssn } = this.formData;
+        const format = (ssnInputVal) => {
+            // Remove all non-dash and non-numerals
+            let formatted = ssnInputVal.replace(/[^\d-]/g, '');
+
+            // Add the first dash if a number from the second group appears
+            formatted = formatted.replace(/^(\d{3})-?(\d{1,2})/, '$1-$2');
+
+            // Add the second dash if a number from the third group appears
+            formatted = formatted.replace(/^(\d{3})-?(\d{2})-?(\d{1,4})/, '$1-$2-$3');
+
+            // Remove misplaced dashes
+            formatted = formatted.split('').filter((val, idx) => val !== '-' || idx === 3 || idx === 6).join('');
+
+            // Enforce max length
+            return formatted.substring(0, 11);
+        };
+
+        ssn.value = format(ssn.value);
+    }
+
+    async handleSubmit(): Promise<void> {
+        this.validateAll({ asTouched: true });
```

Comment:
What's `asTouched` do/imply? I can't tell from the `validateAll` function method body.

Reply:
There are cases of validation logic where we don't show a validation error if the input isn't what we're calling "touched". Such as when a user starts typing and we don't want to show an error until they leave. In the case of submit, we want to validate everything "as touched" since the user is basically saying they're done inputing at that point.

Reply:
Got it, very helpful. Thank you!